### DISCUSSION TITLE
problem with pushing to the github

### DISCUSSION THREAD

**Discussion by arefxv** [Cyfrin/foundry-full-course-cu#2920] [Q&A]
hi!
I pushed my code to my github repo but there are two errors
there's a little red cross in my profile
errors:
  Error: fatal: No url found for submodule path 'git' in .gitmodules
  Error: The process '/usr/bin/git' failed with exit code 128 
  
  wish you could help me

**Comment by EngrPips** [comment]
Hello, I am unsure what could cause this error, when you browse google, what did you find?

**Comment by LordAdhaar** [comment]
> https://confluence.atlassian.com/bbkb/fix-error-fatal-no-url-found-for-submodule-path-in-gitmodules-1188400518.html#:~:text=Essentially%2C%20the%20root%20cause%20of,the%20submodules%20may%20be%20updated.&text=url%2Dof%2Dsource%3A%20is,that%20contains%20the%20submodule's%20files.

This is a great article on the issue you are facing, alternatively

`[submodule "lib/forge-std"]
	path = lib/forge-std
	url = https://github.com/foundry-rs/forge-std
	
[submodule "lib/openzeppelin-contracts"]
	path = lib/openzeppelin-contracts
	url = https://github.com/OpenZeppelin/openzeppelin-contracts

[submodule "lib/chainlink-brownie-contracts"]
	path = lib/chainlink-brownie-contracts
	url = https://github.com/smartcontractkit/chainlink-brownie-contracts
`

This is what my .gitmodules looks like. I don't know which project you are working on or what modules you need. But you can pick whichever module you need and copy paste the [submodule], path and url 

For example if your .gitmodules is empty and you need openzeppelin-contracts, paste this in your .gitmodules file 

`[submodule "lib/openzeppelin-contracts"]
	path = lib/openzeppelin-contracts
	url = https://github.com/OpenZeppelin/openzeppelin-contracts`

**Reply by EngrPips** [reply]
Great heads-up.
